Possible NPE during diagram refresh when a conditional style does not have a predicate

Rerported through [this forum post](https://www.eclipse.org/forums/index.php/m/1833946/):

```
Caused by: java.lang.NullPointerException: null
at org.eclipse.sirius.web.interpreter.ExpressionConverter.convertExpression(ExpressionConverter.java:57) ~[sirius-web-interpreter-0.1.0-SNAPSHOT.jar!/:0.1.0-SNAPSHOT]
at org.eclipse.sirius.web.interpreter.AQLInterpreter.evaluateExpression(AQLInterpreter.java:108) ~[sirius-web-interpreter-0.1.0-SNAPSHOT.jar!/:0.1.0-SNAPSHOT]
at org.eclipse.sirius.web.compat.diagrams.ContainerMappingStyleProvider.apply(ContainerMappingStyleProvider.java:54) ~[sirius-web-compatibility-0.1.0-SNAPSHOT.jar!/:0.1.0-SNAPSHOT]
at org.eclipse.sirius.web.compat.diagrams.ContainerMappingStyleProvider.apply(ContainerMappingStyleProvider.java:36) ~[sirius-web-compatibility-0.1.0-SNAPSHOT.jar!/:0.1.0-SNAPSHOT]
at org.eclipse.sirius.web.diagrams.components.NodeComponent.render(NodeComponent.java:70) ~[sirius-web-diagrams-0.1.0-SNAPSHOT.jar!/:0.1.0-SNAPSHOT]
at org.eclipse.sirius.web.components.BaseRenderer.renderComponent(BaseRenderer.java:200) ~[sirius-web-components-0.1.0-SNAPSHOT.jar!/:0.1.0-SNAPSHOT]
at org.eclipse.sirius.web.components.BaseRenderer.renderElement(BaseRenderer.java:121) ~[sirius-web-components-0.1.0-SNAPSHOT.jar!/:0.1.0-SNAPSHOT]
at org.eclipse.sirius.web.components.BaseRenderer.renderChildren(BaseRenderer.java:170) ~[sirius-web-components-0.1.0-SNAPSHOT.jar!/:0.1.0-SNAPSHOT]
at org.eclipse.sirius.web.components.BaseRenderer.renderInstance(BaseRenderer.java:143) ~[sirius-web-components-0.1.0-SNAPSHOT.jar!/:0.1.0-SNAPSHOT]
at org.eclipse.sirius.web.components.BaseRenderer.renderElement(BaseRenderer.java:117) ~[sirius-web-components-0.1.0-SNAPSHOT.jar!/:0.1.0-SNAPSHOT]
at org.eclipse.sirius.web.components.BaseRenderer.renderComponent(BaseRenderer.java:202) ~[sirius-web-components-0.1.0-SNAPSHOT.jar!/:0.1.0-SNAPSHOT]
at org.eclipse.sirius.web.components.BaseRenderer.renderElement(BaseRenderer.java:121) ~[sirius-web-components-0.1.0-SNAPSHOT.jar!/:0.1.0-SNAPSHOT]
at org.eclipse.sirius.web.diagrams.renderer.DiagramRenderer.render(DiagramRenderer.java:43) ~[sirius-web-diagrams-0.1.0-SNAPSHOT.jar!/:0.1.0-SNAPSHOT]
at org.eclipse.sirius.web.spring.collaborative.diagrams.DiagramService.create(DiagramService.java:71) ~[sirius-web-spring-collaborative-diagrams-0.1.0-SNAPSHOT.jar!/:0.1.0-SNAPSHOT]
at org.eclipse.sirius.web.spring.collaborative.diagrams.DiagramRefreshManager.computeDiagram(DiagramRefreshManager.java:98) ~[sirius-web-spring-collaborative-diagrams-0.1.0-SNAPSHOT.jar!/:0.1.0-SNAPSHOT]
at org.eclipse.sirius.web.spring.collaborative.diagrams.DiagramRefreshManager.refresh(DiagramRefreshManager.java:91) ~[sirius-web-spring-collaborative-diagrams-0.1.0-SNAPSHOT.jar!/:0.1.0-SNAPSHOT]
at org.eclipse.sirius.web.spring.collaborative.diagrams.DiagramEventProcessor.refresh(DiagramEventProcessor.java:104) ~[sirius-web-spring-collaborative-diagrams-0.1.0-SNAPSHOT.jar!/:0.1.0-SNAPSHOT]
at org.eclipse.sirius.web.spring.collaborative.projects.ProjectEventProcessor.lambda$doHandle$3(ProjectEventProcessor.java:208) ~[sirius-web-spring-collaborative-0.1.0-SNAPSHOT.jar!/:0.1.0-SNAPSHOT]
at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183) ~[na:na]
at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177) ~[na:na]
at java.base/java.util.concurrent.ConcurrentHashMap$ValueSpliterator.forEachRemaining(ConcurrentHashMap.java:3605) ~[na:na]
at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484) ~[na:na]
at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474) ~[na:na]
at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150) ~[na:na]
at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173) ~[na:na]
at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[na:na]
at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497) ~[na:na]
at org.eclipse.sirius.web.spring.collaborative.projects.ProjectEventProcessor.doHandle(ProjectEventProcessor.java:207) ~[sirius-web-spring-collaborative-0.1.0-SNAPSHOT.jar!/:0.1.0-SNAPSHOT]
at org.eclipse.sirius.web.spring.collaborative.projects.ProjectEventProcessor.lambda$handle$1(ProjectEventProcessor.java:140) ~[sirius-web-spring-collaborative-0.1.0-SNAPSHOT.jar!/:0.1.0-SNAPSHOT]
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[na:na]
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[na:na]
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[na:na]
```

--- a/backend/sirius-web-interpreter/src/main/java/org/eclipse/sirius/web/interpreter/ExpressionConverter.java
+++ b/backend/sirius-web-interpreter/src/main/java/org/eclipse/sirius/web/interpreter/ExpressionConverter.java
@@ -13,6 +13,7 @@
 package org.eclipse.sirius.web.interpreter;
 
 import java.util.List;
+import java.util.Optional;
 
 /**
  * Used to convert expression written using Sirius specific interpreters such as var: and feature: into proper AQL
@@ -52,7 +53,7 @@ public class ExpressionConverter {
     private static final String ESCAPED_QUOTE = "\\'"; //$NON-NLS-1$
 
     public String convertExpression(String expressionBody) {
-        String processedExpression = expressionBody;
+        String processedExpression = Optional.ofNullable(expressionBody).orElse(""); //$NON-NLS-1$
 
         if (processedExpression.startsWith(VAR_PREFIX)) {
             // var:variableName -> aql:variableName

--- a/backend/sirius-web-interpreter/src/test/java/org/eclipse/sirius/web/interpreter/AQLInterpreterTestCases.java
+++ b/backend/sirius-web-interpreter/src/test/java/org/eclipse/sirius/web/interpreter/AQLInterpreterTestCases.java
@@ -92,4 +92,20 @@ public class AQLInterpreterTestCases {
         assertThat(interpreter).isNotNull();
     }
 
+    @Test
+    public void testNullExpressionEvaluatesToTrue() {
+        AQLInterpreter interpreter = new AQLInterpreter(List.of(), List.of(EcorePackage.eINSTANCE));
+        Result result = interpreter.evaluateExpression(Map.of(), null);
+        assertThat(result).isNotNull();
+        assertThat(result.asBoolean()).contains(Boolean.TRUE);
+    }
+
+    @Test
+    public void testEmptyExpressionEvaluatesToTrue() {
+        AQLInterpreter interpreter = new AQLInterpreter(List.of(), List.of(EcorePackage.eINSTANCE));
+        Result result = interpreter.evaluateExpression(Map.of(), ""); //$NON-NLS-1$
+        assertThat(result).isNotNull();
+        assertThat(result.asBoolean()).contains(Boolean.TRUE);
+    }
+
 }

--- a/backend/sirius-web-interpreter/src/test/java/org/eclipse/sirius/web/interpreter/ExpressionConverterTestCases.java
+++ b/backend/sirius-web-interpreter/src/test/java/org/eclipse/sirius/web/interpreter/ExpressionConverterTestCases.java
@@ -88,4 +88,11 @@ public class ExpressionConverterTestCases {
         String aqlExpression = new ExpressionConverter().convertExpression(varExpression);
         assertThat(aqlExpression).isEqualTo("aql:arg0"); //$NON-NLS-1$
     }
+
+    @Test
+    public void testNullExpressionConvertedToEmptyStringExpression() {
+        String expression = null;
+        String aqlExpression = new ExpressionConverter().convertExpression(expression);
+        assertThat(aqlExpression).isEqualTo("aql:''"); //$NON-NLS-1$
+    }
 }

